TrackletAlgorithm/TrackHandler.cc

```diff
@@ -1,82 +1,81 @@
 #include "TrackHandler.h"
 
-bool TrackHandler::compareTrack(TrackStruct& trk, TrackStruct& masterTrk, unsigned int& matchFound, unsigned int mergeCondition){
-  // compare the two tracks, masterTrack and trk
-  LOOP_CompareTrackBarrel:
+bool TrackHandler::compareTrack(const TrackStruct &trk, TrackStruct &masterTrk, unsigned int& matchFound, unsigned int mergeCondition){
+  // Compare the two tracks, masterTrack and trk
+  #pragma HLS array_partition variable=matchesFoundBarrel complete dim=0
+  #pragma HLS array_partition variable=matchesFoundDisk complete dim=0
+  #pragma HLS array_partition variable=masterTrk._barrelStubArray complete dim=0
+  #pragma HLS array_partition variable=trk._barrelStubArray complete dim=0
+  #pragma HLS array_partition variable=masterTrk._diskStubArray complete dim=0
+  #pragma HLS array_partition variable=trk._diskStubArray complete dim=0
+  LOOP_CompareBarrelStubs:
   for (unsigned int barrelStubNum = 0; barrelStubNum < NBarrelStub; barrelStubNum++){
     #pragma HLS unroll
-    auto masterBarrelStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(masterTrk._barrelStubArray[barrelStubNum][0].range(kBarrelStubIndexSizeMSB, kBarrelStubIndexSizeLSB));
-    auto inputBarrelStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(trk._barrelStubArray[barrelStubNum][0].range(kBarrelStubIndexSizeMSB, kBarrelStubIndexSizeLSB));
-
-    if((masterBarrelStubIndex == inputBarrelStubIndex) && (masterBarrelStubIndex > 0)){
-      matchesFoundBarrel[barrelStubNum][0] = 1;
-    } else {
-      matchesFoundBarrel[barrelStubNum][0] = 0;
-    }
+    ap_uint<TrackFitType::kTFStubIndexSize> masterBarrelStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(masterTrk._barrelStubArray[barrelStubNum][0].range(kBarrelStubIndexSizeMSB, kBarrelStubIndexSizeLSB));
+    ap_uint<TrackFitType::kTFStubIndexSize> inputBarrelStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(trk._barrelStubArray[barrelStubNum][0].range(kBarrelStubIndexSizeMSB, kBarrelStubIndexSizeLSB));
+    
+    ((masterBarrelStubIndex == inputBarrelStubIndex) && (masterBarrelStubIndex > 0)) ? matchesFoundBarrel[barrelStubNum][0] = 1 : matchesFoundBarrel[barrelStubNum][0] = 0;
```

Comment:
This can be written slightly more compactly, with less duplication:
```suggestion
    matchesFoundBarrel[barrelStubNum][0] = ((masterBarrelStubIndex == inputBarrelStubIndex) && (masterBarrelStubIndex > 0)) ?  1 : 0;
```